edb/edgeql/compiler/inference/multiplicity.py

```diff
@@ -570,7 +570,8 @@ def _infer_for_multiplicity(
     scope_tree: irast.ScopeTreeNode,
     ctx: inf_ctx.InfCtx,
 ) -> inf_ctx.MultiplicityInfo:
-    itset = ir.iterator_stmt
+    # XXX: This is wrong.
```

Comment:
Why, and does this need fixing?

Reply:
This just looks at the first iterator and disregards the rest. We'd have to implement multiplicity inference for Cartesian product:

- infer multiplicity for each of the product elements,
- if any elements is EMPTY, multiplicity of the product is also EMPTY,
- if any elements is DUPLICATE, multiplicity of the product is also DUPLICATE,
- otherwise it's UNIQUE.